technologies/job/informatica-powercenter-soap/tasks/instanceActions.js

```diff
@@ -0,0 +1,135 @@
+  try {
+    console.log('STOP INSTANCE:', instance);
+    const client = await getDataIntegrationClientAuthenticated(job.featuresValues);
+
+    await client.stopTaskAsync({
+      DIServiceInfo: {
+        ServiceName: job.featuresValues.service.label,
+      },
+      FolderName: job.featuresValues.folder.label,
+      WorkflowName: job.featuresValues.workflow.label,
+      RequestMode: job.featuresValues.requestMode.id,
+      TaskInstancePath: job.featuresValues.task.label,
+    });
+
+    return Response.success();
+  } catch (error) {
+    return Response.error(ERRORS_MESSAGES.FAILED_TO_STOP_JOB_ERROR, { error });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance status.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getStatus = async ({ job, instance }) => {
+  try {
+    console.log('GET STATUS INSTANCE:', instance);
+    const client = await getDataIntegrationClientAuthenticated(job.featuresValues);
+
+    const res = await client.getTaskDetailsAsync({
+      DIServiceInfo: {
+        ServiceName: job.featuresValues.service.label,
+      },
+      FolderName: job.featuresValues.folder.label,
+      WorkflowName: job.featuresValues.workflow.label,
+      TaskInstancePath: job.featuresValues.task.label,
+    });
+
+    const resBody = getResponseBodyFromSOAPRequest(res);
+
+    if (resBody && resBody.length > 0 && resBody[0]) {
+      return Response.success(
+        JOB_STATES[resBody[0].TaskRunStatus || resBody[0].WorkflowRunStatus]
+        || JobStatus.AWAITING
+      );
+    }
+
+    return Response.empty();
+  } catch (error) {
+    return Response.error(ERRORS_MESSAGES.FAILED_TO_GET_STATUS_ERROR, { error });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance logs.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    const client = await getDataIntegrationClientAuthenticated(job.featuresValues);
+
+    const res = await client.getSessionLogAsync({
+      DIServiceInfo: {
+        ServiceName: job.featuresValues.service.label,
+      },
+      FolderName: job.featuresValues.folder.label,
+      WorkflowName: job.featuresValues.workflow.label,
+      TaskInstancePath: job.featuresValues.task.label,
+      Timeout: DEFAULT_TIMEOUT,
+    });
+
+    const resBody = getResponseBodyFromSOAPRequest(res);
+
+    if (resBody && resBody.length > 0 && resBody[0]) {
+      const logs = resBody[0].Buffer;
+      const logsLines = logs.split('\n');
+
+      return Response.success(logsLines.map((logLine) => Log(logLine, Stream.STDOUT, null)));
```

Comment:
```suggestion
      return Response.success(logsLines.map((logLine) => Log(logLine, Stream.STDOUT)));
```

No need to give `null`, this call contains a default value.

Reply:
```suggestion
      return Response.success(logsLines.map((logLine) => Log(logLine)));
```

Same for the second parameter. If there is only STDOUT, no need to give it.

> https://github.com/saagie/sdk/blob/master/packages/sdk/src/sdk/Log.js#L10